### ISSUE TITLE
add_site_option in RestrictedFunctions grouping 

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#370]
## What problem would the enhancement address for VIP?

Add `add_site_option()` to the RestrictedFunctions sniff, as we already have `update_site_option()` and `delete_site_option()` for consistent grouping.  We'll also need to silence it on the WordPress-VIP-Go ruleset.

## What code should be reported as a violation?

```php
add_site_option( 'blahblah' );
```

## What could should *not* be reported as a violation?

```php
add_option();
```
